### PR TITLE
Simplified the file filter predication code generation.

### PR DESCRIPTION
**Pull request by stuartskelton**
Ok, I may be a little guilty of golfing a little here, but this removes the brain ache of trying to keep track of what has been set or not.

The only reason I could get away with this is because all the checks were ANDs.

### CODE REVIEW

**Comment by stevan** [comment]
Looks mostly good to me, I am adding a few small comments in the code for discussion.

File: lib/App/Critique/Command/collect.pm [hunk 1]
```diff
@@ -179,98 +179,25 @@ sub generate_file_predicate {
...
-    else {
-        $predicate = sub () { 1 };
-    }
+    my $c = $session->perl_critic;
+
+    # lets build an array of code_ref filters, that will be use to filter
+    # the files, the code refs assume the params will be $path,$rel.
+    my @filters = (sub { return 1 });
+    push @filters, sub { return $_[1] !~ /$filter/} if $filter;
+    push @filters, sub { return $_[1] =~ /$match/ } if $match ;
```
Comment (stevan) [inline lib/App/Critique/Command/collect.pm]: I think the order of `filter` and `match` here is reversed, if you look at the old code, `match` always precedes `filter'.

File: lib/App/Critique/Command/collect.pm [hunk 2]
```diff
@@ -179,98 +179,25 @@ sub generate_file_predicate {
...
-    #    1   |   1   |      0       # collect with filter and match
-    #    1   |   1   |      1       # collect with filter match and no violations
-    #    1   |   0   |      1       # collect with filter and no violations
-    # ------------------------------#
-    #    0   |   1   |      0       # collect with match
-    #    0   |   1   |      1       # collect with match and no violations
-    # ------------------------------#
-    #    0   |   0   |      1       # collect with no violations
-    # ------------------------------#
-
```
Comment (stevan) [inline lib/App/Critique/Command/collect.pm]: I would like to retain this table in the new version, just as documentation.

**Comment by stevan** [comment]
Okay, comments added, one question and one request.

File: lib/App/Critique/Command/collect.pm (hunk 1 above)
Comment (stuartskelton) [inline lib/App/Critique/Command/collect.pm]: you are very correct

**Comment by stuartskelton** [comment]
I have corrected the filters.

and:

I have also added back the table, however with a little change in the ordering, so the stages descend in their filters binary value, I think it makes it a little more readable, but I am happy to change it back if you disagree.

**Comment by stuartskelton** [comment]
Thinking about it too, I should probably switch Cols 1 and 2. too as that would fit with the logic.

**Comment by stevan** [comment]
Yeah, switching cols 1 and 2 would be good, I agree, after that, this patch is good, thanks!

**Comment by stuartskelton** [comment]
I have amended the last commit.

**Pull request merged**